### DISCUSSION TITLE
add support for sum columns horizontally

### DISCUSSION THREAD

**Discussion by djouallah** [duckdb/duckdb#13815] [Ideas]
I am using this new functionality and I think it is a game changer and changed the way I see SQL 

```
xx = duckdb.sql(f" select *, greatest(*columns(*exclude(query))) as max from result  ")
xx
```

now give me sum and addition :)  no list_aggregate workaround,  you people started this  🥇 and there is no going back

**Comment by Tishj** [comment]
To do that efficiently I think you'll want to perform a PIVOT of some sort first

**Comment by djouallah** [comment]
@Tishj nah, no pivot,  the whole idea is not to pivot, you guys can do something behind the scene :)

**Comment by soerenwolfers** [comment]
I believe you already know, but I'm gonna state it here for completeness: 

You can already do 
```sql
SELECT
  list_sum(list_value(*COLUMNS(*)))
```

While there might be a chance that a `n`-ary `sum` or `add` will be implemented, which would allow `sum(*COLUMNS(*))`, I don't think this is the way to happiness: The next thing you'll want to do is a horizontal cumulative sum, and I don't think an `n`-ary `cumulative_sum: Float^n -> Float[n]` will ever be implemented (although I might be wrong, I'm not a duckdb dev).

Instead, if you accept the `list_value(*COLUMNS(*))` "inconvenience", you can do pretty much everything your heart desires, using the `list_aggregate` / `list_transform` / `list_reduce` functions (although cumulative operations will still be difficult until https://github.com/duckdb/duckdb/discussions/11541 is addressed)

**Comment by djouallah** [comment]
@soerenwolfers thank you, make sense
